Added a bangladesh flag

garvitgupta13 opened the pull request:
I have created a new folder "flags" in "assets", containing a Bangladesh flag. Please let me know should I also add others too.

garvitgupta13 committed:
Merge pull request #1 from firstcontributions/master

Syncing forked repo

garvitgupta13 committed:
Added bangladesh flag

garvitgupta13 committed:
Added a flag folder containing all flag images

garvitgupta13 committed:
Added 8 flags of bg br cn cz de dk eg es

Roshanjossey commented:
Hi @garvitgupta13, thank you for making this change.

I've some suggestions.

1. Let's not add flag assets in our repo. I'm thinking about moving the assets we have here outside to reduce the size of this repo. It would be easier for us if we have to maintain less (including assets)

2. The flags you added are not of a uniform style
   
![image](https://user-images.githubusercontent.com/8488446/93892534-e99bba80-fcec-11ea-996b-16a51427cee5.png)
![image](https://user-images.githubusercontent.com/8488446/93892640-0a641000-fced-11ea-87f3-37f82f932ce1.png)
![image](https://user-images.githubusercontent.com/8488446/93892686-1780ff00-fced-11ea-9d63-4f822b660510.png)
![image](https://user-images.githubusercontent.com/8488446/93892734-249dee00-fced-11ea-8e82-924b30af3ca6.png)

These four have four different styles. That's not great.

**How I think this can be improved**

I found this repo with good assets for flags and it's actively maintained https://github.com/hjnilsson/country-flags. 

Over the network, most of them took less than a KB which is good.

Example:

```<img src="https://cdn.staticaly.com/gh/hjnilsson/country-flags/master/svg/bd.svg" width="22">```
<img src="https://cdn.staticaly.com/gh/hjnilsson/country-flags/master/svg/bd.svg" width="22">

Roshanjossey commented on README.md:
```diff
@@ -14,16 +14,16 @@ Reading articles & watching tutorials can help, but what's better than actually
-[:bangladesh:](translations/README.bn.md)
-[🇧🇬](translations/README.bg.md)
-[🇧🇷](translations/README.pt_br.md)
+[<img src="assets/flags/bd.png" width="22">](translations/README.bn.md)
```
```suggestion
[<img src="https://cdn.staticaly.com/gh/hjnilsson/country-flags/master/svg/bd.svg" width="22">](translations/README.bn.md)
```

Feel free to ask if you have questions

garvitgupta13 replied:
I also agree with you @Roshanjossey .  Should I do this with all others flags.

Roshanjossey replied:
Yeah, go ahead.

garvitgupta13 committed:
Added flags image of BD BG BR CN CZ DE DK EG ES

garvitgupta13 committed:
Replaced all the country codes with their flags

garvitgupta13 committed:
Added flags image of FR AL GL which were missed in previous commit

garvitgupta13 committed:
Removing the flags folder from assets

garvitgupta13 commented:
As per your suggestion, I have replaced all the country codes with their flags

![image](https://user-images.githubusercontent.com/70805691/93975864-b2cdaf00-fd95-11ea-9506-3e01b2b202fb.png)

Roshanjossey commented:
Good job @garvitgupta13

I've two suggestions 

1. Please make the same changes to https://github.com/firstcontributions/first-contributions/blob/master/translations/Translations.md
2. Nepal flag is too big because of it's proportions. Could you please change the width to 15px?
![image](https://user-images.githubusercontent.com/8488446/93985351-3d59e280-fd85-11ea-801d-db2c3b8f7c86.png)


I also have one question that I don't have an answer for. 

![image](https://user-images.githubusercontent.com/8488446/93985464-68443680-fd85-11ea-9270-44cbc6a02575.png)

Japanese and South Korean flags don't look very good in this background
before
![image](https://user-images.githubusercontent.com/8488446/93985971-0932f180-fd86-11ea-8878-ebef5ae6f635.png)
after
![image](https://user-images.githubusercontent.com/8488446/93985858-e6a0d880-fd85-11ea-86cb-9132eca08626.png)


Flags with white as their base matches with the background and you don't see it. I thought about giving a different background for this section but I don't think that'll work. Maybe I'll think of some weird workaround in the future.

garvitgupta13 committed:
Reduced Nepal's flag width from 22 to 15

garvitgupta13 committed:
Changed JP and KR flags

garvitgupta13 committed:
Added image of JP and KR flags in assets

garvitgupta13 committed:
Replacing the korean flag

garvitgupta13 committed:
Replaced the country codes with their flags in Translations.md

garvitgupta13 commented:
As per your suggestions:

1.  Reduced Nepal's flag width to 15
2.  Replaced the country codes with their flags in Translations.md
3.  Replaced the JP and KR flags

Roshanjossey commented:
Awesome. Thank you for making these changes @garvitgupta13. Merging

garvitgupta13 commented:
Really happy to know that

Roshanjossey commented:
I've put all the flags in <kbd>  tag to get a grey background now. Also put Japanese and South Korean flags back. It's not perfect but I think it works.

garvitgupta13 commented:
After this change, the user can easily find japan and south Korea flags irrespective of their background. That's exactly what we wanted.
